src/libraries/System.Private.CoreLib/src/System/TimeOnly.cs

```diff
@@ -0,0 +1,961 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Globalization;
+
+namespace System
+{
+    /// <summary>
+    /// Represents a time of day, as would be read from a clock, within the range 00:00:00 to 23:59:59.9999999.
+    /// </summary>
+    public readonly struct TimeOnly : IComparable, IComparable<TimeOnly>, IEquatable<TimeOnly>, IFormattable, ISpanFormattable
+    {
+        // represent the number of ticks map to the time of the day. 1 ticks = 100-nanosecond in time measurements.
+        private readonly long _ticks;
+
+        // MinTimeTicks is the ticks for the midnight time 00:00:00.000 AM
+        private const long MinTimeTicks = 0;
+
+        // MaxTimeTicks is the max tick value for the time in the day. It is calculated using DateTime.Today.AddTicks(-1).TimeOfDay.Ticks.
+        private const long MaxTimeTicks = 863_999_999_999;
+
+        /// <summary>
+        /// Represents the smallest possible value of TimeOnly.
+        /// </summary>
+        public static TimeOnly MinValue => new TimeOnly((ulong) MinTimeTicks);
+
+        /// <summary>
+        /// Represents the largest possible value of TimeOnly.
+        /// </summary>
+        public static TimeOnly MaxValue => new TimeOnly((ulong) MaxTimeTicks);
```

Comment:
Nit: we typically don't have spaces in the middle of casts (I actually thought we enabled a stylecop rule about that, but I guess not)
```suggestion
        public static TimeOnly MinValue => new TimeOnly((ulong)MinTimeTicks);

        /// <summary>
        /// Represents the largest possible value of TimeOnly.
        /// </summary>
        public static TimeOnly MaxValue => new TimeOnly((ulong)MaxTimeTicks);
```

Reply:
will do.